crates/ra_tt/src/lib.rs

```diff
@@ -56,21 +56,47 @@ pub struct Subtree {
 
 #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
 pub enum Delimiter {
```

Comment:
I think this is ok, but I wonder if

```
pub struct Subtree {
    pub delimiter: Option<Delimiter>,
    pub token_trees: Vec<TokenTree>,
}

struct Delimiter {
  id: TokenId,
  kind: DelimiterKind,
}

struct DelimiterKind {
  Paranthesis, Brace, Bracket,
}
```

will lead to cleaner code?

Reply:
Yeah, I think we should go with the representation above. As a first step, it might be a good idea to send a separate PR which refactors current repr to 

```rust
#[derive(Debug, Clone, PartialEq, Eq, Hash)]
pub struct Subtree {
    pub delimiter: Option<Delimiter>,
    pub token_trees: Vec<TokenTree>,
}

#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
pub enum Delimiter {
    Parenthesis,
    Brace,
    Bracket,
}
```

Reply:
Sure, submitted #2551